roachtest: improve flake detection for typeorm test

rafiss opened the pull request:
fixes https://github.com/cockroachdb/cockroach/issues/100705
fixes https://github.com/cockroachdb/cockroach/issues/103182
fixes https://github.com/cockroachdb/cockroach/issues/103175
fixes https://github.com/cockroachdb/cockroach/issues/103172

Release note: None

rafiss committed:
roachtest: improve flake detection for typeorm test

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/107979)
<!-- Reviewable:end -->

rimadeodhar commented:
Could you provide a better commit message on why we are skipping the test using a regex match? Also, is ignoreList/blockList not an option here?

rafiss committed:
roachtest: fix knex flake

Updating the version fixes an issue with the test where javascript
dependencies were not installed correctly.

Release note: None

rafiss commented:
good question - i added a comment instead. the reason is that we don't have a way to parse javascript test results programatically (or at least, not the ones from this library). so instead, we have to do custom checks inside the code. the typeorm test is actually similar too. tftr!

bors r+

rafiss commented:
bors r+
